### DISCUSSION TITLE
HTTP Status 401 Fobbidden

### DISCUSSION THREAD

**Discussion by romerma** [dasniko/testcontainers-keycloak#71] [Q&A]
Hello 👋🏽,

first, thank you so much dasniko for this project.

I am trying to execute some integration tests using the keycloak config, which i am using for production. Unfortunately i'm getting  401 status codes and i don't know why. 

Thanks in advance.

<img width="1624" alt="image" src="https://user-images.githubusercontent.com/21131549/160689567-34b5fb73-afc3-49a9-8eb9-79886b47d89a.png">

**Comment by dasniko** [comment]
Hi @uriberma 

It's hard to guess, as you only provide some log output but nothing else.
From the screenshot, the last warn message tells you, that you are trying to authenticate with user `admin` in realm `quarkus` but the user can't be found. So, most likely, you don't have a user `admin` in your `quarkus` realm!?

💡 ProTipp: Please don't paste screenshots of your log or other text-based files in issues/discussions, as they are only images, can't be searched and are hard to read on small screens. GitHub provides good text formatting possibilities to past the actual text here. Thanks.

**Comment by romerma** [comment]
Hi @dasniko,

thank you very much for your prompt answer and the ProTipp 👍🏽 

Yes, the user admin is present in the quarkus realm and that's why I don't understand.

I found that the user is added to the master realm and not to the dedicated realm was I have created for this app, could be that the problem?

Beste Grüße

**Reply by dasniko** [reply]
I don't know what the cause for your issue is. As already mentioned, the only thing that can be assumed from your provided information is, that there is no user named "admin" in your "quarkus" realm. That's what the error event says. There's no more information available.
